--- mod_time/mod_time.c	2004-05-12 14:57:06.000000000 -0400
+++ mod_time/mod_time.c.oden	2009-02-27 15:24:47.000000000 -0500
@@ -188,7 +188,7 @@ static void build_group_arrays(pool *p, 
   /* Nothing there...punt.
    */
   if (!pw) {
-    if (!name || !(pw = (struct passwd *) auth_getpwnam(p, name))) {
+    if (!name || !(pw = (struct passwd *) pr_auth_getpwnam(p, name))) {
       *gids = xgids;
       *groups = xgroups;
       return;
@@ -197,16 +197,16 @@ static void build_group_arrays(pool *p, 
 
   /* Populate the first group name.
    */
-  if ((gr = auth_getgrgid(p, pw->pw_gid)) != NULL)
+  if ((gr = pr_auth_getgrgid(p, pw->pw_gid)) != NULL)
     *((char **) push_array(xgroups)) = pstrdup(p, gr->gr_name);
 
-  auth_setgrent(p);
+  pr_auth_setgrent(p);
 
   /* This is where things get slow, expensive, and ugly.
    * Loop through everything, checking to make sure we haven't already added
    * it.  This is why we have getgroups() and company.
    */
-  while ((gr = auth_getgrent(p)) != NULL && gr->gr_mem)
+  while ((gr = pr_auth_getgrent(p)) != NULL && gr->gr_mem)
     for (gr_mem = gr->gr_mem; *gr_mem; gr_mem++) {
       if (strcmp(*gr_mem, pw->pw_name) == 0) {
         *((int *) push_array(xgids)) = (int) gr->gr_gid;
@@ -634,7 +634,7 @@ static pr_time_map_t *get_user_cmd_times
   if (!session.user) {
     /* Populate the pw struct, check for a NULL return value. */
 
-    if ((pw = auth_getpwnam(cmd->tmp_pool, user)) == NULL)
+    if ((pw = pr_auth_getpwnam(cmd->tmp_pool, user)) == NULL)
       return NULL;
 
     time_user = pstrdup(session.pool, pw->pw_name);
@@ -1603,7 +1603,7 @@ static int time_display_file(const char 
        "%R", (session.c && session.c->remote_name ?  session.c->remote_name : "(unknown)"),
        "%L", main_server->ServerFQDN,
        "%U", user,
-       "%u", session.ident_user,
+       "%u", session.user,
        "%M", mesg_max,
        "%N", mesg_cur,
        "%E", main_server->ServerAdmin,
