--- mod_autohost/mod_autohost.c	2010-02-09 10:56:17.000000000 -0500
+++ mod_autohost/mod_autohost.c.oden	2011-12-06 07:20:09.233539720 -0500
@@ -407,13 +407,8 @@ static void autohost_postparse_ev(const
             pr_netaddr_get_ipstr(main_server->addr), ports[i], strerror(errno));
           continue;
         }
-
-        /* Create a listening socket for this port.  The given server list
-         * turns out not to be needed or even used by pr_inet_create_conn().
-         * Figures.
-         */
-        listener = pr_inet_create_conn(autohost_pool, autohost_server_list,
-          -1, main_server->addr, ports[i], FALSE);
+	/* http://forums.proftpd.org/smf/index.php?topic=5356.0 */
+        listener = pr_inet_create_conn(autohost_pool, -1, main_server->addr, ports, FALSE);
         if (listener == NULL) {
           (void) pr_log_writefile(autohost_logfd, MOD_AUTOHOST_VERSION,
             "error opening new listening socket for port %d: %s", ports[i],
